pkg/kn/commands/configuration_edit_flags.go

```diff
@@ -82,6 +80,38 @@ func (p *ConfigurationEditFlags) Apply(config *servingv1alpha1.ConfigurationSpec
 	return nil
 }
 
+// Replace resets previous ConfigurationSpec and configures the provided one
+func (p *ConfigurationEditFlags) Replace(config *servingv1alpha1.ConfigurationSpec) error {
```

Comment:
I think the replace command should work more like `create` — don't bother blanking things and setting them to new things, just create a new service resource as if you were creating it anew.

Reply:
@sixolet : I think I didn't get this. 

>  just create a new service resource as if you were creating it anew.

How does this is different than `service create` then?
Or
What are the scenarios  a dev would use `service replace` over `service create` ?

Reply:
create doesnt allow to "update" resource if it doesnt exist. though i could see extending create to allow that functionality via a flag instead of having a dedicated command.

Reply:
So `kn service replace` is similar to `kubectl apply` when using a resource file, right ?  (except maybe that `kubectl apply` also allows a create)

* It correlates over the resource `name`
* It replaces the existing resource but only if there are not immutable fields which can't be removed

In this simple case, it's just a kind of combination of `delete` followed by a `create`. This works nicely for top-level resources. I think we should follow the same semantic to not allow a "replace" (aks "apply") if there is an immutable field set (though I currently can't think about one in the `service`).

For possibly auto-created objects like revisions or configuration, we should check that this by verifying the `ownerReference` and forbid a replace is that case (only for non-auto generated objects I would allow a replace).

------

@sixolet Having a second look at the example in the issue description, I'm now not sure anymore whether I get it, too, when you say

> I think the replace command should work more like create — don't bother blanking things and setting them to new things, just create a new service resource as if you were creating it anew.

and how it applies to e.g.

```
kn service replace s1 --env KEY1=NEW_VALUE1 --env NEW_KEY2=NEW_VALUE2
```

In this case, where should the `image` come from then ? Shouldn't it be just taken from the original service? And if so, this is really not different to a`service update` ? Or asked otherwise, which fields would you expect to be blank / resetted after this operation ? This question applies to the other usage examples in the issue description, too.

@cppforlife I agree that if this command all about a "set" of a resource with all fields to set given as as parameters, its probably a better choice to use a `kn service create --force` to allow an overwrite of an existing resource (with the minor caveat that this would not validate that a resource already exists like a `kn service replace` probably would do)

Reply:
Replace should require `--image`, and should always remove all currently-set env vars, along with limits, etc, anything not metnioned explicitly.

Feel free to impl a --force flag on create instead, I'm not super picky about the wording.

Reply:
Cool, I think in order to keep the UX surface small (and to avoid confusions between `create`, `replace` and `update`) I opt for a `--force` option to create as `create` already has to check for all mandatory parameters for creating a service (like the image).

Reply:
@rhuss @cppforlife @sixolet : Okay, I have added `--force` flag to `service create` command instead, PTAL.